thirdparty/github.go

```diff
@@ -835,6 +836,31 @@ func GithubUserInOrganization(ctx context.Context, token, requiredOrganization,
 	return isMember, err
 }
 
+// AppAuthorizedForOrg returns true if the given app name exists in the org's installation list,
+// and has permission to write to pull requests. Returns an error if the app name exists but doesn't have permission.
+func AppAuthorizedForOrg(ctx context.Context, token, requiredOrganization, name string) (bool, error) {
+	httpClient := getGithubClientRetry(token, "AppAuthorizedForOrg")
+	defer utility.PutHTTPClient(httpClient)
+
+	client := github.NewClient(httpClient)
+	// We can assume no org will have more than 100 installations.
+	installations, _, err := client.Organizations.ListInstallations(ctx, requiredOrganization, &github.ListOptions{PerPage: 100})
```

Comment:
Is there a way for us to maybe cache this information so that we don't have to check every time. Like maybe cache it in the project and only check if the information is not cached? 
or do we expect this to change often?

Reply:
We can maybe add some logic to off boarding scripts if we are worried about someone having access after leaving the company.

Reply:
I would avoid caching because it won't respond to GitHub-side app changes. If for example, someone deletes a bot or changes its permissions, we shouldn't be allowed to authorize patches for it.